core/drivers/stm32_rng.c

```diff
@@ -149,7 +150,7 @@ TEE_Result stm32_rng_read(uint8_t *out, size_t size)
 {
 	TEE_Result rc = 0;
 	uint32_t exceptions = 0;
-	vaddr_t rng_base = io_pa_or_va(&stm32_rng->base);
+	vaddr_t rng_base = io_pa_or_va(&stm32_rng->base, size);
```

Comment:
size to fix: it's not the right size to check.
I would prefer here `1` as RNG IOmem interface is known to not cross page boundaries.
```
	vaddr_t rng_base = io_pa_or_va(&stm32_rng->base, 1);
```

Reply:
Ok, so I suppose that in drivers/hi16xx_rng.c size in phys_to_virt() should be changed to 1 too.